pkg/admission/customresourcevalidation/oauth/validate_openid.go

```diff
@@ -0,0 +1,36 @@
+package oauth
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	configv1 "github.com/openshift/api/config/v1"
+	crvalidation "github.com/openshift/origin/pkg/admission/customresourcevalidation"
+	"github.com/openshift/origin/pkg/cmd/server/apis/config/validation/common"
+)
+
+func ValidateOpenIDIdentityProvider(provider *configv1.OpenIDIdentityProvider, identityProvider configv1.IdentityProvider, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if provider == nil {
+		allErrs = append(allErrs, field.Required(fieldPath, ""))
+		return allErrs
+	}
+
+	allErrs = append(allErrs, ValidateOAuthIdentityProvider(provider.ClientID, provider.ClientSecret, fieldPath.Child("provider"))...)
+
+	if provider.Issuer != strings.TrimRight(provider.Issuer, "/") {
```

Comment:
@enj I originally implemented due to the request in https://github.com/openshift/cluster-authentication-operator/blob/master/pkg/operator2/idp.go#L237 but that makes this single spot stand out as the single one with this requirement.

Is there a reason why we couldn't continue trimming trailing `/` and adding the oidc discover endpoint instead?

Reply:
Hm, this is actually also a part of check in the request header IdP, I guess we could leave it here then.